docs/DeveloperGuide.md

```diff
@@ -1,34 +1,182 @@
+    * [Clearing the list](DeveloperGuide.md#clearing-the-list)
+    * [Saving data](DeveloperGuide.md#saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Setting up
 
+## Design
 
-## Product scope
-### Target user profile
+### Architecture
+
+### UI component
+
+The UI consists of `TextUi` class. It acts as a middleware between users' input
+and `Parser` class under **Logic Component**.
+
+The `UI` component,
+
+* Reads user commands using `Scanner` class.
+* Pass raw user commands into `Parser` class under **Logic Component**.
+* After the **Logic Component** has executed the function, it will then return `CommandOutput` object
+  which contains the outcome of the execution, fail or succeed.
+
+### Logic component
+
+![](images/LogicComponentStructure.png?raw=true "Structure of Logic Component")
+*Figure #. Structure of the Logic Component*
+
+**API** : [`seedu.duke.logic`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/logic) package (*TO BE CREATED LATER*)
+
+The Logic component,
+1. uses the `Parser` class to parse the user input obtained by the `TextUi` class of `UI`.
+2. This results in a specific `Command` object to be returned from the `Parser` class.
+3. Depending on the type of the `Command` object, its corresponding command execution will be called by `Duke` of `UI`.
+4. The command execution can affect the `Model` (e.g. clearing the list). 
+5. The result of the command execution is encapsulated as a `CommandOutput` object which is passed back to the `TextUi`.
+6. In addition, the `CommandOutput` object can also instruct the `TextUi` to perform certain actions, such as displaying the list to the user.
+
+Given below is the Sequence Diagram for interactions within the `Logic` component for the `parseCommand("clear)"` API call.
+
+![](images/LogicComponentSequenceDiagram.png?raw=true "Logic component Sequence Diagram for clear")
+*Figure #. Interactions Inside the Logic Component for the `clear` Command*
+
+### Model component
+
+**API** : [`seedu.duke.person`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/person) package
+
+
+<img src="images/ModelComponentStructure.png" width="600" height="600" />
+
+*Figure #. Structure of the Model Component*
+
+The Model component,
+
+* contains a `Person` class which represents a person who checks in/out.
+* contains a `TrackingList` class which uses an ArrayList to keep track of all the `Person` objects who have currently checked in/out.
+* contains a `PersonLog` class which uses a HashMap to permanently store all the `Person` objects who have checked in before.
+
+A `Person` object contains:
+* an `Id` object
+* a `Name` object
+* a `Phone` object
+
+
+### Storage component
+**API** : [`seedu.duke.storage`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/storage) package
```

Comment:
Inconsistent formatting, Tracking List Encoder should be written as TrackingListEncoder